slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -156,29 +278,27 @@ void SLAMServiceImpl::CreateMap() {
     int end_scan_number = int(file_list.size());
     for (int i = this->starting_scan_number; i < end_scan_number; i++) {
         if (!b_continue_session) return;
-
-        auto measurement =
-            mapBuilder.GetDataFromFile(this->path_to_data, initial_file, i);
-        if (measurement.ranges.size() > 0) {
-            trajectory_builder->AddSensorData(kRangeSensorId.id, measurement);
-            int num_nodes = mapBuilder.map_builder_->pose_graph()
+        int num_nodes;
+        {
+            std::lock_guard<std::mutex> lk(map_builder_mutex);
+            auto measurement = map_builder.GetDataFromFile(this->path_to_data,
+                                                           initial_file, i);
+            if (measurement.ranges.size() > 0) {
+                trajectory_builder->AddSensorData(kRangeSensorId.id,
+                                                  measurement);
+                num_nodes = map_builder.map_builder_->pose_graph()
                                 ->GetTrajectoryNodes()
                                 .size();
 
-            auto local_poses = GetLocalPoses(mapBuilder);
-            if (local_poses.size() > 0) {
-                auto latest_local_pose = local_poses.back();
-                cartographer::transform::Rigid3d global_pose =
-                    GetGlobalPose(mapBuilder, trajectory_id, latest_local_pose);
-                myfile << "global_pose: " << global_pose.DebugString()
-                       << std::endl;
-            }
-
-            if ((num_nodes >= this->starting_scan_number &&
-                 num_nodes < this->starting_scan_number + 3) ||
-                num_nodes % this->picture_print_interval == 0) {
-                PaintMap(mapBuilder.map_builder_, this->path_to_map + "/images",
-                         std::to_string(num_nodes));
+                auto local_poses = map_builder.GetLocalSlamResultPoses();
```

Comment:
What's the difference between GetLocalPoses() and GetLocalSlamResultPoses()?

Reply:
`GetLocalPoses` was simply a function within `server_functions.cc` that called `GetLocalSlamResultPoses()` and returned those results. Since we have `map_builder` now within the slam service, we can just call the function directly. I removed `GetLocalPoses` by deleting `server_functions.cc`.